internal/core/src/segcore/StreamReduce.cpp

```diff
@@ -520,17 +520,30 @@ StreamReducerHelper::GetSearchResultDataSlice(int slice_index) {
     auto nq_begin = slice_nqs_prefix_sum_[slice_index];
     auto nq_end = slice_nqs_prefix_sum_[slice_index + 1];
 
-    int64_t result_count =
-        merged_search_result->topk_per_nq_prefix_sum_[nq_end] -
-        merged_search_result->topk_per_nq_prefix_sum_[nq_begin];
-
     auto search_result_data =
         std::make_unique<milvus::proto::schema::SearchResultData>();
     // set unify_topK and total_nq
     search_result_data->set_top_k(slice_topKs_[slice_index]);
     search_result_data->set_num_queries(nq_end - nq_begin);
     search_result_data->mutable_topks()->Resize(nq_end - nq_begin, 0);
 
+    int64_t result_count = 0;
```

Comment:
can we add unit test for that case?

the test coverage should be much higher than what we current have

Reply:
I will take this